packages/devtools/lib/src/memory/memory.dart

```diff
@@ -185,6 +228,205 @@ class MemoryScreen extends Screen with SetStateMixin {
+      final String classSelected = heapPopupList.highlightedItem;
+      _selectClass(classSelected);
+    }
+
+    // Done with the popup.
+    heapPopup.hide();
+  }
+
+  Future<void> _selectInstanceByHashCode(String objectHashCode) async {
+    // There's an instances table up.
+    final Table<Object> instanceTable = tableStack.last;
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check each instance looking to find a particular object.
+      final evalResult = await evaluate(instance.objectRef, 'hashCode');
+      if (evalResult != null && evalResult.valueAsString == objectHashCode) {
+        // Found the object select the instance.
+        instanceTable.selectByIndex(row, scrollBehavior: 'auto');
+        return;
+      }
+      row++;
+    }
+
+    framework.toast('Unable to find instance [$objectHashCode]',
+        title: 'Error');
+  }
+
+  void selectClassInstance(String className, String objectHashCode) {
+    // Remove selection in class list.
+    tableStack.first.clearSelection();
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) {
+      if (!tableStack.first.hasSelection) {
+        // Wait until the class list has no selection.
+        timer.cancel();
+      }
+    });
+
+    // Select the class (don't record this select in memory history). The
+    // memoryPath will be added by NavigationState.inboundSelect - see below.
+    _selectClass(className, false);
+
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) async {
+      // Wait until the class has been selected, 2 lists (class and instances
+      // for the class exist) and the instances list has data.
+      if (tableStack.first.hasSelection &&
+          tableStack.length == 2 &&
+          tableStack.last.data.isNotEmpty) {
+        timer.cancel();
+
+        await _selectInstanceByHashCode(objectHashCode);
+      }
+    });
+  }
+
+  void selectClassAndInstanceInField(
+    String className,
+    String field,
+    String hashCode,
+  ) async {
+    fromMemoryHover = true;
+
+    // Remove selection in class list.
+    tableStack.first.clearSelection();
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) {
+      if (!tableStack.first.hasSelection) {
+        // Wait until the class list has no selection.
+        timer.cancel();
+      }
+    });
+
+    // Select the class (don't record this select in memory history). The
+    // memoryPath will be added by NavigationState.inboundSelect - see below.
+    _selectClass(className, false);
+
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) async {
+      // Wait until the class has been selected, 2 lists (class and instances
+      // for the class exist) and the instances list has data.
+      if (tableStack.first.hasSelection &&
+          tableStack.length == 2 &&
+          tableStack.last.data.isNotEmpty) {
+        timer.cancel();
```

Comment:
so far all the logic in this method is repeated. Can we consolidate using a helper?

Reply:
Added _isClassSelectedAndInstancesRead